src/main/java/org/gridsuite/modification/server/entities/equipment/creation/BranchCreationEntity.java

```diff
@@ -99,23 +101,34 @@ private void assignAttributes(BranchCreationInfos branchCreationInfos) {
         voltageLevelId2 = branchCreationInfos.getVoltageLevelId2();
         busOrBusbarSectionId1 = branchCreationInfos.getBusOrBusbarSectionId1();
         busOrBusbarSectionId2 = branchCreationInfos.getBusOrBusbarSectionId2();
-        if (branchCreationInfos.getCurrentLimits1() != null) {
-            currentLimits1 = new CurrentLimitsEntity(branchCreationInfos.getCurrentLimits1());
-        } else {
-            currentLimits1 = null;
-        }
-        if (branchCreationInfos.getCurrentLimits2() != null) {
-            currentLimits2 = new CurrentLimitsEntity(branchCreationInfos.getCurrentLimits2());
-        } else {
-            currentLimits2 = null;
-        }
+        currentLimits1 = assignCurrentLimits(branchCreationInfos.getCurrentLimits1(), currentLimits1);
+        currentLimits2 = assignCurrentLimits(branchCreationInfos.getCurrentLimits2(), currentLimits2);
         connectionDirection1 = branchCreationInfos.getConnectionDirection1();
         connectionName1 = branchCreationInfos.getConnectionName1();
         connectionDirection2 = branchCreationInfos.getConnectionDirection2();
         connectionName2 = branchCreationInfos.getConnectionName2();
+        selectedOperationalLimitsGroupId1 = branchCreationInfos.getSelectedOperationalLimitsGroupId1();
```

Comment:
getActiveOperationalLimits ?

Reply:
This could have been a good name but I use the iidm terminology. Cf LineAttributes.java in powsybl network store.